modules/core/src/main/java/org/apache/ignite/internal/processors/cache/persistence/wal/FileWriteAheadLogManager.java

```diff
@@ -3621,4 +3723,73 @@ static long minWalArchiveSize(DataStorageConfiguration dsCfg) {
     @Override public void startAutoReleaseSegments() {
         segmentAware.startAutoReleaseSegments();
     }
+
+    private class WalSegmentAsyncCloser extends GridWorker {
+        final ConcurrentLinkedQueue<IgniteBiTuple<FileWriteHandle, Boolean>> tasks = new ConcurrentLinkedQueue<>();
```

Comment:
Why don't you use a blocking queue? You wouldn't have to park the thread manually. Just a question, I'm not suggesting anything

Reply:
If you reffering to the `ArrayBlockingQueue` then I believe it synchronized on same mutex on adding and extracting element. LinkedQueue on contrary doesn't use mutexes.